lib/spack/spack/environment/environment.py

```diff
@@ -597,6 +626,10 @@ def regenerate(self, concretized_root_specs):
             tty.debug("View at %s does not need regeneration." % self.root)
             return
 
+        if use_renameat2:
+            if os.path.isdir(new_root):
+                shutil.rmtree(new_root)
+
```

Comment:
add some comments on this one to say what happens when we're configured to use symlinks but we find a `renameat2` directory and vice/versa.